server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -306,61 +287,52 @@ private void registerRequestHandler(DynamicActionRegistry dynamicActionRegistry)
-            }
-            if (!extensionIdMap.isEmpty()) {
-                logger.info("Loaded all extensions");
-            }
-        } else {
-            logger.warn("Extensions.yml file is not present.  No extensions will be loaded.");
-        }
-    }
-
     /**
      * Loads a single extension
      * @param extension The extension to be loaded
      */
-    private void loadExtension(Extension extension) throws IOException {
-        if (extensionIdMap.containsKey(extension.getUniqueId())) {
-            logger.info("Duplicate uniqueId " + extension.getUniqueId() + ". Did not load extension: " + extension);
-        } else {
-            try {
-                DiscoveryExtensionNode discoveryExtensionNode = new DiscoveryExtensionNode(
-                    extension.getName(),
-                    extension.getUniqueId(),
-                    new TransportAddress(InetAddress.getByName(extension.getHostAddress()), Integer.parseInt(extension.getPort())),
-                    new HashMap<String, String>(),
-                    Version.fromString(extension.getOpensearchVersion()),
-                    Version.fromString(extension.getMinimumCompatibleVersion()),
-                    extension.getDependencies()
-                );
+    public void loadExtension(Extension extension) throws IOException {
+        try {
+            validateExtension(extension);
+            DiscoveryExtensionNode discoveryExtensionNode = new DiscoveryExtensionNode(
+                extension.getName(),
+                extension.getUniqueId(),
+                new TransportAddress(InetAddress.getByName(extension.getHostAddress()), Integer.parseInt(extension.getPort())),
+                new HashMap<String, String>(),
+                Version.fromString(extension.getOpensearchVersion()),
+                Version.fromString(extension.getMinimumCompatibleVersion()),
+                extension.getDependencies()
+            );
+            extensionIdMap.put(extension.getUniqueId(), discoveryExtensionNode);
+            extensionSettingsMap.put(extension.getUniqueId(), extension);
+            logger.info("Loaded extension with uniqueId " + extension.getUniqueId() + ": " + extension);
+        } catch (IOException e) {
+            throw e;
+        } catch (IllegalArgumentException e) {
+            throw e;
+        }
 
-                extensionIdMap.put(extension.getUniqueId(), discoveryExtensionNode);
-                extensionSettingsMap.put(extension.getUniqueId(), extension);
-                logger.info("Loaded extension with uniqueId " + extension.getUniqueId() + ": " + extension);
-            } catch (OpenSearchException e) {
-                logger.error("Could not load extension with uniqueId " + extension.getUniqueId() + " due to " + e);
-            } catch (IllegalArgumentException e) {
-                throw e;
-            }
+    }
+
+    private boolean validateExtension(Extension extension) throws IOException {
```

Comment:
Is it worth combining this into the ExtensionDependency logic or vice versa? Seems like we may want to just check required fields once whereas this check UniqueID and Version a second time (plus other stuff).

Reply:
We should not combine this `ExtensionDependency` logic because that's a different model class and has it's own parameters like `uniqueId` and `version`. And they should have a different validation of their own.

Reply:
Method returns a `boolean` but really it's either returning `true` or throwing an exception. Could change return to `void`.